soc/arm/microchip_mec/Kconfig

```diff
@@ -11,6 +12,212 @@ config SOC_FAMILY
 	string
 	default "microchip_mec"
 
+menuconfig MCHP_MEC_UNSIGNED_HEADER
+	bool "Create an unsigned output binary with MCHP MEC binary header"
+	help
+	  On Microchip MEC series chip, the ROM code loads firmware image from flash
+	  to RAM using a TAG to locate a Header which specifies the location and
+	  size of the firmware image. Enable this to invoke the mec_spi_gen tool
+	  which generates an SPI image with TAG, Header, and firmware binary. This
+	  tool does not produce a signed image which can be authenticated by the
+	  Boot-ROM. Use the full Microchip SPI image generator program for
+	  authentication and all other Boot-ROM loader features. Refer to the MCHP
+	  EVB boards for an example.
+
+if MCHP_MEC_UNSIGNED_HEADER
+
+config MCHP_MEC_IMAGE_OUTPUT_BIN
+	bool "Build Microchip MEC binary in BIN format"
+	default y
+	help
+	  Build a "raw" binary zephyr/zephyr.mchp.bin in the build directory.
+	  The name of this file can be customized with CONFIG_KERNEL_BIN_NAME.
+
+config MCHP_MEC_HEADER_CHIP
+	string
+	default "mec152x" if SOC_SERIES_MEC1501X
+	default "mec172x" if SOC_SERIES_MEC172X
+
+choice MCHP_MEC_HEADER_SPI_FREQ_MHZ_CHOICE
+	prompt "Clock rate to use for SPI flash"
+	default MCHP_MEC_HEADER_SPI_FREQ_MHZ_12
+	help
+	  This selects the SPI clock frequency that will be used for loading
+	  firmware binary from flash to RAM.
+
+config MCHP_MEC_HEADER_SPI_FREQ_MHZ_12
+	bool "SPI flash clock rate of 12 MHz"
+
+config MCHP_MEC_HEADER_SPI_FREQ_MHZ_16
+	bool "SPI flash clock rate of 16 MHz"
+
+config MCHP_MEC_HEADER_SPI_FREQ_MHZ_24
+	bool "SPI flash clock rate of 24 MHz"
+
+config MCHP_MEC_HEADER_SPI_FREQ_MHZ_48
+	bool "SPI flash clock rate of 48 MHz"
+endchoice
+
+config MCHP_MEC_HEADER_SPI_FREQ_MHZ
+	int
+	default 12 if MCHP_MEC_HEADER_SPI_FREQ_MHZ_12
+	default 25 if MCHP_MEC_HEADER_SPI_FREQ_MHZ_16
+	default 24 if MCHP_MEC_HEADER_SPI_FREQ_MHZ_24
+	default 48 if MCHP_MEC_HEADER_SPI_FREQ_MHZ_48
+
+choice MCHP_MEC_HEADER_SPI_READ_MODE_CHOICE
+	prompt "Reading mode used by the SPI flash"
+	default MCHP_MEC_HEADER_SPI_READ_MODE_FAST
+	help
+	  This sets the reading mode that can be used by the SPI flash.
+	  Reading modes supported are normal, fast, dual, and quad.
+
+config MCHP_MEC_HEADER_SPI_READ_MODE_NORMAL
+	bool "SPI flash operates full-duplex with frequency (< 25 MHz)"
+
+config MCHP_MEC_HEADER_SPI_READ_MODE_FAST
+	bool "SPI flash operates full-duplex with fast reading mode"
+
+config MCHP_MEC_HEADER_SPI_READ_MODE_DUAL
+	bool "SPI flash operates with dual data reading mode"
+
+config MCHP_MEC_HEADER_SPI_READ_MODE_QUAD
+	bool "SPI flash operates with quad data reading mode"
+endchoice
```

Comment:
nit: add a space before endchoice (seems like that's how it's commonly done, also on the other instances below)

Reply:
fixed